fendermint/actors/blobs/shared/src/state.rs

```diff
@@ -86,22 +91,32 @@ pub struct Account {
     /// The chain epoch of the last debit.
     pub last_debit_epoch: ChainEpoch,
     /// Credit approvals to other accounts from this account, keyed by receiver.
-    pub approvals_to: HashMap<String, CreditApproval>,
+    pub approvals_to: CreditApprovals,
```

Comment:
This is the crux of this PR. Remove the `HashMap` to use [`CreditApprovals`](https://github.com/recallnet/ipc/pull/541/files#diff-732ecaf2caff3ea98ad38ab0d514481f98a542238bdd0a10faded80f1b6f1941R434-R466) that is implemented using a `hamt`